internal/store/device.go

```diff
@@ -930,3 +931,38 @@ func (s *DeviceStore) CountByOrgAndStatus(ctx context.Context, orgId *uuid.UUID,
 	}
 	return results, nil
 }
+
+// PrepareDevicesAfterRestore sets the waitingForConnectionAfterRestore annotation
+// on all devices, clears their lastSeen timestamps, and sets status summary using efficient SQL
+func (s *DeviceStore) PrepareDevicesAfterRestore(ctx context.Context) (int64, error) {
+	db := s.getDB(ctx)
+
+	// Use raw SQL for efficient bulk update that preserves existing annotations
+	// and properly handles the status JSON structure using || operator for merging
+	sql := `
+		UPDATE devices 
+		SET 
+			annotations = COALESCE(annotations, '{}'::jsonb) || jsonb_build_object($1::text, 'true'),
+			status = CASE 
+				WHEN status IS NOT NULL THEN 
+					(status - 'lastSeen') || jsonb_build_object('summary', jsonb_build_object('status', $2::text, 'info', $3::text))
+				ELSE 
+					jsonb_build_object('summary', jsonb_build_object('status', $2::text, 'info', $3::text))
+			END,
+			resource_version = COALESCE(resource_version, 0) + 1
+		WHERE deleted_at IS NULL 
+			AND (annotations->>'device-controller/awaitingReconnect' != 'true' OR annotations->>'device-controller/awaitingReconnect' IS NULL)
```

Comment:
1. Why is this necessary?  Why not update all devices regardless of the annotation's value?
2. In addition to deleted devices, we should also skip decommissioned devices.